test/integration/controller/jobset_controller_test.go

```diff
@@ -666,6 +677,550 @@ var _ = ginkgo.Describe("JobSet controller", func() {
+								Succeeded: 0,
+								Failed:    0,
+								Active:    0,
+								Suspended: 1,
+							},
+						})
+					},
+				},
+				{
+					jobUpdateFn: func(jobList *batchv1.JobList) {
+						readyJob(&jobList.Items[0])
+					},
+				},
+				{
+					checkJobSetState: func(js *jobset.JobSet) {
+						matchJobSetReplicatedStatus(js, []jobset.ReplicatedJobStatus{
+							{Name: "replicated-job-b",
+								Ready:     0,
+								Succeeded: 0,
+								Failed:    0,
+								Active:    0,
+								Suspended: 0,
+							},
+							{
+								Name:      "replicated-job-a",
+								Ready:     1,
+								Succeeded: 0,
+								Failed:    0,
+								Active:    0,
+								Suspended: 0,
+							},
+						})
+					},
+					checkJobSetCondition: testutil.JobSetStartupPolicyNotFinished,
+				},
+				{
+					jobUpdateFn: func(jobList *batchv1.JobList) {
+						readyJob(&jobList.Items[1])
+						readyJob(&jobList.Items[2])
+						readyJob(&jobList.Items[3])
+					},
+				},
+				{
+					checkJobSetState: func(js *jobset.JobSet) {
+						matchJobSetReplicatedStatus(js, []jobset.ReplicatedJobStatus{
+							{Name: "replicated-job-b",
+								Ready:     3,
+								Succeeded: 0,
+								Failed:    0,
+								Active:    0,
+								Suspended: 0,
+							},
+							{
+								Name:      "replicated-job-a",
+								Ready:     1,
+								Succeeded: 0,
+								Failed:    0,
+								Active:    0,
+								Suspended: 0,
+							},
+						})
+					},
+					checkJobSetCondition: testutil.JobSetStartupPolicyComplete,
+				},
+			},
+		}),
+		ginkgo.Entry("startupPolicy; create jobs in order", &testCase{
+			makeJobSet: func(ns *corev1.Namespace) *testing.JobSetWrapper {
+				return testJobSet(ns).
+					StartupPolicy(&jobset.StartupPolicy{
+						StartupPolicyOrder: jobset.InOrder,
+					})
+			},
+			skipCreationCheck: true,
+			updates: []*update{
+				{
+					// First update
+					// Replicated-Job-A should be created.
+					// Startup Policy Condition is set
+					checkJobCreation: func(js *jobset.JobSet) {
+						expectedStarts := 1
+						gomega.Eventually(testutil.NumJobs, timeout, interval).WithArguments(ctx, k8sClient, js).Should(gomega.Equal(expectedStarts))
+					},
+					checkJobSetState: func(js *jobset.JobSet) {
+						matchJobSetReplicatedStatus(js, []jobset.ReplicatedJobStatus{
+							{Name: "replicated-job-b",
+								Ready:     0,
+								Succeeded: 0,
+								Failed:    0,
+								Active:    0,
+								Suspended: 0,
+							},
+							{
+								Name:      "replicated-job-a",
+								Ready:     0,
+								Succeeded: 0,
+								Failed:    0,
+								Active:    0,
+								Suspended: 0,
+							},
+						})
+					},
+					checkJobSetCondition: testutil.JobSetStartupPolicyNotFinished,
+				},
+				{
+					// Second update
+					// Set Replicated-Job-A to ready
+					jobUpdateFn: func(jobList *batchv1.JobList) {
+						readyJob(&jobList.Items[0])
+					},
+					checkJobCreation: func(js *jobset.JobSet) {
+						expectedStarts := 1
```

Comment:
shouldn't this be more than 1? it should be 4 since the first replicatedJob is ready and at this point we should expect that the second replicatedJob jobs are created (additional 3)

Reply:
I notice that it takes an update for this to occur. I think it is 4 in the next test case.

Reply:
This is confusing though, lets move it to the next step (to line 907) and check against 4.

Reply:
I mostly used that function as if I don't specify it, I end up checking the default job one. ie it verifies against 4.